PixelDungeon/src/main/java/com/watabou/pixeldungeon/windows/WndJournal.java

```diff
@@ -30,14 +30,25 @@
 import com.watabou.pixeldungeon.ui.Icons;
 import com.watabou.pixeldungeon.ui.ScrollPane;
 import com.watabou.pixeldungeon.ui.Window;
+import com.watabou.pixeldungeon.utils.GLog;
+import com.watabou.pixeldungeon.windows.elements.LabeledTab;
+import com.watabou.pixeldungeon.windows.elements.Tab;
 
 import java.util.Collections;
 
-public class WndJournal extends Window {
+public class WndJournal extends WndTabbed {
+
+	private static final int LEVEL_ITEM_HEIGHT	= 18;	// Height of a level entry
+	private static final int LOGBOOK_ITEM_HEIGHT = 12;	// Height of a log book entry
 
-	private static final int ITEM_HEIGHT	= 18;
-	
 	private static final String TXT_TITLE	= Game.getVar(R.string.WndJournal_Title);
+	private static final String TXT_LEVELS	= Game.getVar(R.string.WndJournal_Levels);
+	private static final String TXT_LOGBOOK	= Game.getVar(R.string.WndJournal_Logbook);
+
+	private Text       txtTitle;
+	private ScrollPane list;
+
+	private static boolean showLevels = true;	// Indicates which tab is visible
```

Comment:
Can we do not use this field, but instead move tab specific functionality to respective tab class, and then just pass required calls to WndTabbed.selected ?

Reply:
Hm this sounds good. I tried to mimic the WndCatalogus behavior, but I had a feeling that it's not the best way. I'll work on this.

Reply:
@Mikhael-Danilov, I need clarification. Do you want to remove the `showLevels` flag from `WndJournal.java`, but leave the `ListItem` class unchanged? Or are you suggesting 2 separate classes, like `ListLevelItem` and `ListLogItem`? Thanks.

Reply:
2 separate classes of course.